llpc/patch/llpcPatchDescriptorLoad.cpp

```diff
@@ -509,8 +527,35 @@ Value* PatchDescriptorLoad::LoadDescriptor(
         }
         else
         {
-            auto pDescOffset = ConstantInt::get(Type::getInt32Ty(*m_pContext), descOffset);
-            auto pDescSize   = ConstantInt::get(Type::getInt32Ty(*m_pContext), descSize, 0);
+            llvm::Value* pDescOffset = nullptr;
+            if (!m_pPipelineState->GetBuilderContext()->BuildingRelocatableElf())
+            {
+                pDescOffset = ConstantInt::get(Type::getInt32Ty(*m_pContext), descOffset);
+            }
+            else
+            {
+                IRBuilder<> builder(*m_pContext);
+                builder.SetInsertPoint(pInsertPoint);
+                std::stringstream ssRelocName;
+                ssRelocName << "doff_" << descSet << "_" << binding;
+                if (nodeType1 == ResourceMappingNodeType::DescriptorSampler)
+                {
+                    ssRelocName << "_s";
+                }
+                else if (nodeType1 == ResourceMappingNodeType::DescriptorResource)
+                {
+                    ssRelocName << "_r";
+                }
+                else
+                {
+                    ssRelocName << "_b";
+                }
```

Comment:
We should start a markdown document that we can add to the LLPC repo that documents how the relocations work.  We can get feedback from AMD on how they want it documented.

Reply:
Ack.